Desktop_Interface/uartstyledecoder.cpp

```diff
@@ -180,27 +192,24 @@ bool uartStyleDecoder::jitterCompensationProcedure(double baudRate, unsigned cha
 }
 
 //Basically scaffolding to add character maps for other modes (5 bit, for example).
-void uartStyleDecoder::decodeDatabit(int mode){
-    char tempchar;
-    switch(mode){
+char uartStyleDecoder::decodeDatabit(int mode, short symbol) const
+{
+    switch(mode)
+	{
         case 5:
-            tempchar = decode_baudot(currentUartSymbol);
+            return decodeBaudot(symbol);
             break;
         case 8:  //8-bit ASCII;
-            tempchar = currentUartSymbol;
+            return symbol;
             break;
         default:
             qDebug() << "uartStyleDecoder::decodeDatabit is failing...";
+			return -1; // Garbage
     }
-    if (parityCheckFailed)
-    {
-        serialBuffer->insert("\n<ERROR: Following character contains parity error>\n");
-        parityCheckFailed = false;
-    }
-    serialBuffer->insert(tempchar);
 }
 
-char uartStyleDecoder::decode_baudot(short symbol){
+char uartStyleDecoder::decodeBaudot([[maybe_unused]] short symbol) const
```

Comment:
I wouldn't suppress this warning here - there should be a switch/case statement here enumerating all the different values for baudot encoding.  It's just an empty function for now as this feature is unimplemented and so constantly getting a reminder every time I compile is a good thing.  :)

Reply:
I feel like unused argument is not the right warning though. Maybe adding a #warning would be better (it's not standard afaik but i think most compilers support it)?

Reply:
I'll remove the anotation for now